models/agent_helpers.go

```diff
@@ -193,6 +193,51 @@ func FindAgentsByIDs(q *reform.Querier, ids []string) ([]*Agent, error) {
 	return res, nil
 }
 
+// FindAgentsOnServiceWithType find agents by specified serviceID and serviceType.
+func FindAgentsOnServiceWithType(q *reform.Querier, serviceID string, serviceType ServiceType) ([]*Agent, error) {
```

Comment:
How about name this func `FindCredsForService` and make it self-sufficient? It can accept only `serviceID` and return `DBConfig`.

In our case, we will select service twice, but it's not so big overhead.